glean-core/src/lib_unit_tests.rs

```diff
@@ -893,7 +893,8 @@ fn records_io_errors() {
 
     // Writing the ping file should fail.
     let submitted = glean.internal_pings.metrics.submit(&glean, None);
-    assert!(submitted.is_err());
+    // But the return value is still Ok because we enqueue the ping anyway.
+    assert!(submitted.is_ok());
```

Comment:
Should we extend this test to ensure that the uploader will see both pings (this one that's not written to disk and the one that is written to disk)?

Fine if done in a followup